### ISSUE TITLE
How to work with other super-resolution tools ?

### ISSUE THREAD

**Issue by Michaelwhite34** [yangxy/GPEN#64]
Suppose I have a folder of enhanced images of the same name, how to just paste enhanced face to these images without scaling original images ?

**Comment by Michaelwhite34** [comment]
> The scaling in face_enhancement.py is optional. I did it to better detect the faces due to the faces in the original images are too small.
> 
> If you want to combine a super-resolution method with GPEN, I suggest the following steps:
> 
> 1. employing the SR method on the original image to get SR result;
> 2. resizing the original image to the size of the SR output;
> 3. performing GPEN on the resized original image;
> 4. pasting enhanced faces back to the SR result.
> 
> The last step can be readily achieved by reformulating `img = cv2.convertScaleAbs(img*(1-full_mask) + full_img*full_mask) ` in face_enhancement.py to `img = cv12.convertScaleAbs(img_sr*(1-full_mask) + full_img*full_mask)`.
> 
> I will update a combined version soon.

Thanks for the reply, where can I disable the scaling ?

**Comment by Michaelwhite34** [comment]
> > Also,I notice that there will always be "face detail" on noisy photos.When the training comes out,will it be easier to train a model that generates cleaner and soft face like remini ?
> 
> did you find anything that comes closer to remini? @Michaelwhite34

At this point I would say gpen is as good as remini as long as your input is not messed up like small size gifs.If you use cleaner dataset to train gpen, it will look closer to remini.

**Comment by haiderasad** [comment]
@Michaelwhite34 have yet to try gpen but uptill now  GFPGAN gives the best results in comparison with remini , also heard about DFDNet but not yet implementaed

**Comment by Michaelwhite34** [comment]
> @Michaelwhite34 have yet to try gpen but uptill now GFPGAN gives the best results in comparison with remini , also heard about DFDNet but not yet implementaed

I tried all these,gpen is still best.GFPgan is too sharp,DFDNet is not high quality enough.

**Comment by haiderasad** [comment]
@Michaelwhite34 that is very interesting saves me the time, thanks for the insight

**Comment by haiderasad** [comment]
@Michaelwhite34 can you please share the GPEN-BFR-2048 pretrained model as its been taken down

**Comment by Michaelwhite34** [comment]
> @Michaelwhite34 can you please share the GPEN-BFR-2048 pretrained model as its been taken down

That's weird,you can ask the author why it was taken down ?

**Comment by haiderasad** [comment]
yes, can you send the file at my email, haiderasadbhutta@gmail.com @Michaelwhite34

**Comment by farasas** [comment]
> > @Michaelwhite34 can you please share the GPEN-BFR-2048 pretrained model as its been taken down
> 
> That's weird,you can ask the author why it was taken down ? Never mind,I saw commerical reasons.



> yes, can you send the file at my email, [haiderasadbhutta@gmail.com](mailto:haiderasadbhutta@gmail.com) @Michaelwhite34

could you share it with me?

elena7gzz@gmail.com

**Comment by BbChip0103** [comment]
@Michaelwhite34 @farasas @haiderasad 

If you have GPEN-BFR-2048 checkpoint, can you share it to me also? Please save me.
bbchip13@gmail.com

**Comment by yannis-tml** [comment]
Hi, could you share me the model by email too please? 
@farasas @BbChip0103  @Michaelwhite34  

tumelinyannis@gmail.com

**Comment by zimenglan-sysu-512** [comment]
@Michaelwhite34 @farasas @haiderasad

If you have GPEN-BFR-2048 checkpoint, can you share it to me also? Please save me.
[dongdk.sysu@foxmail.com](mailto:dongdk.sysu@foxmail.com)
